File: superstitioApi/java/superstitioapi/powers/interfaces/invisible/InvisiblePower_InvisibleRemovePowerEffectPatch.java
package superstitioapi.powers.interfaces.invisible;

import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction;
import javassist.CannotCompileException;
import javassist.expr.ExprEditor;
import javassist.expr.MethodCall;

import java.util.ArrayList;

public class InvisiblePower_InvisibleRemovePowerEffectPatch {

    @SpirePatch(clz = RemoveSpecificPowerAction.class, method = "update")
    public static class HideExpireText {
        public static ExprEditor Instrument() {
            return new ExprEditor() {
                public void edit(final MethodCall m) throws CannotCompileException {
                    if (m.getClassName().equals(ArrayList.class.getName()) && m.getMethodName().equals("add")) {
                        m.replace("if (!(" + InvisiblePower_InvisibleRemovePowerEffect.class.getName() +
                                ".shouldInvisibleRemovePowerEffect(removeMe))) {$_ = $proceed($$);}");
                    }
                }
            };
        }
    }
}

File: superstitioApi/java/superstitioapi/powers/interfaces/invisible/InvisiblePower_InvisibleRemovePowerEffectPatch.kt
package superstitioapi.powers.interfaces.invisible

import com.evacipated.cardcrawl.modthespire.lib.SpireInstrumentPatch
import com.evacipated.cardcrawl.modthespire.lib.SpirePatch2
import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction
import javassist.CannotCompileException
import javassist.expr.ExprEditor
import javassist.expr.MethodCall

object InvisiblePower_InvisibleRemovePowerEffectPatch
{
    @SpirePatch2(clz = RemoveSpecificPowerAction::class, method = "update")
    object HideExpireText
    {
        @SpireInstrumentPatch
        @JvmStatic
        fun Instrument(): ExprEditor
        {
            return object : ExprEditor()
            {
                @Throws(CannotCompileException::class)
                override fun edit(m: MethodCall)
                {
                    if (m.className == ArrayList::class.qualifiedName && m.methodName == "add")
                    {
                        m.replace(
                            "if (!(" + InvisiblePower_InvisibleRemovePowerEffect.Companion::class.qualifiedName +
                                    ".shouldInvisibleRemovePowerEffect(removeMe))) {\$_ = \$proceed($$);}"
                        )
                    }
                }
            }
        }
    }
}
